src/pages/patientView/clinicalInformation/ClinicalInformationMutationalSignatureTable.tsx

```diff
@@ -0,0 +1,94 @@
+import {ClinicalAttribute, ClinicalData} from "../../../shared/api/generated/CBioPortalAPI";
+import LazyMobXTable, {Column} from "shared/components/lazyMobXTable/LazyMobXTable";
+import styles from './style/mutationalSignatureTable.module.scss';
+import {SHOW_ALL_PAGE_SIZE} from "../../../shared/components/paginationControls/PaginationControls";
+import {IMutationalSignature} from "../../../shared/model/MutationalSignature";
+import convertSamplesData, {IConvertedSamplesData} from "./lib/convertSamplesData";
+import {getMutSigPercentage} from "../../../shared/lib/FormatUtils";
+import _ from 'lodash';
+import {labelMobxPromises} from "mobxpromise";
+
+export interface IClinicalInformationMutationalSignatureTableProps {
+    data: IMutationalSignature[];
+    showTitleBar: boolean;
+    cssClass?:string;
+    showFilter?:boolean;
+    showCopyDownload?:boolean;
+}
+
+class MutationalSignatureTable extends LazyMobXTable<IMutationalSignatureRow> {}
+
+interface IMutationalSignatureRow {
+    mutationalSignatureId:string;
+    sampleValues:{
+        [uniqueSampleKey: string]: { //each element in the row will contain data about exposure and confidence
+            value: number
+            confidence: number
+        }
+    };
+}
+
+export function prepareDataForTable(mutationalSignatureData: IMutationalSignature[]):IMutationalSignatureRow[] {
+
+    const tableData: IMutationalSignatureRow[] = [];
+
+    //group data by mutational signature
+    //[{id: mutationalsignatureid, samples: [{}, {}]}]
+    let sampleInvertedDataByMutationalSignature:Array<any> = _(mutationalSignatureData)
+        .groupBy(mutationalSignatureSample => mutationalSignatureSample.mutationalSignatureId)
+        .map((mutationalSignatureSampleData, mutationalSignatureId)  => ({id: mutationalSignatureId, samples: mutationalSignatureSampleData}))
+        .value();
+
+    for (const mutationalSignature of sampleInvertedDataByMutationalSignature){
+        let mutationalSignatureRowForTable: IMutationalSignatureRow = {mutationalSignatureId:"", sampleValues: {}};
+        mutationalSignatureRowForTable.mutationalSignatureId = mutationalSignature.id;
+        for (const sample of mutationalSignature.samples){
+            mutationalSignatureRowForTable.sampleValues[sample.uniqueSampleKey] =
+                {value: sample.value,
+                confidence: sample.confidence};
+        }
+        tableData.push(mutationalSignatureRowForTable);
+    }
+    return tableData;
+}
+
+export default class ClinicalInformationMutationalSignatureTable extends React.Component<IClinicalInformationMutationalSignatureTableProps, {}> {
+
+    public render() {
+        const uniqueSamples = _.map(_.uniqBy(this.props.data, "uniqueSampleKey"), (uniqSample => ({id: uniqSample.uniqueSampleKey})));
+        const tableData = prepareDataForTable(this.props.data);
+        const firstCol = 'mutationalSignatureId';
+        const columns:Column<IMutationalSignatureRow>[] = [{
```

Comment:
its better to set these columns up outside the `render` method, for two reasons:

(1) (less important here bc its not computationally expensive, but in general something to watch out for) so that you avoid unnecessarily recomputing things on every render. theres nothing in this object that actually needs to be recomputed on every render, these can be a constant and in fact it could really be declared outside the class.

(2) (this is the main reason) react decides whether to rerender things based on if their props have changed. this way, the props to `MutationalSignatureTable` will change every time, because a new javascript object for columns is passed every time. this will lead to unnecessary renders.